README.md

```diff
@@ -0,0 +1,130 @@
+==============
+
+Mender is an open source over-the-air (OTA) software updater for embedded Linux devices. Mender comprises a client running at the embedded device, as well as a server that manages deployments across many devices.
+
+This repository contains mender-convert, which is used to convert pre-built disk images (Debian, Ubuntu, Raspbian, etc) to a Mender compatible image by restructuring partition table and injecting the necessary files.
+
+For a full list of tested devices and images please visit [Mender Hub](https://hub.mender.io/c/board-integrations/debian-family). If your device and image combination is not listed as supported, this does not necessarily mean that it will not work, it probably just means that no one has tested and reported it back and usually only small tweaks are necessary to get this running on your device.
+
+![Mender logo](https://mender.io/user/pages/resources/06.digital-assets/mender.io.png)
+
+## Getting started
+
+To start using Mender, we recommend that you begin with the Getting started
+section in [the Mender documentation](https://docs.mender.io/).
+
+For more detailed information about `mender-convert` please visit the
+[Debian family](https://docs.mender.io/2.0/artifacts/debian-family) section in
+[the Mender documentation](https://docs.mender.io/).
+
+## Docker environment for mender-convert
+
+In order to be able to manipulate and create filesystem and disk images,
+mender-convert has a few dependencies, and their version and name vary between
+Linux distributions. Here is an example of how to install the dependencies on a
+Debian based distribution:
+
+```
+sudo apt-get install -y binutils xz-utils file rsync parted e2fsprogs xfsprogs \
+                        pigz dosfstools wget git make bmap-tools
+```
+
+To make using mender-convert easier, a reference setup using a Ubuntu 19.04
+Docker container is provided.
+
+You need to [install Docker Engine](https://docs.docker.com/install) to use
+this environment.
+
+
+### Build the mender-convert container image
+
+To build a container based on Ubuntu 18.04 with all required dependencies for
+mender-convert, copy this directory to your workstation and change the current
+directory to it.
+
+Then run
+
+```bash
+./docker-build
+```
+
+This will create a container image which you can use to run `mender-convert`
+without polluting your host environment with the necessary dependencies.
+
+
+### Use the mender-convert container image
+
+Create a `input` directory in the root of where you cloned this repository.
+
+```bash
+mkdir input
+```
+
+Move your raw disk image into `input/`, e.g.
+
+```bash
+mv ~/Downloads/2019-04-08-raspbian-stretch-lite.img input/2019-04-08-raspbian-stretch-lite.img
+```
+
+Bootstrap the demo rootfs overlay that is configured to connect to
+https://hosted.mender.io with polling intervals set appropriately for
+demonstration purposes:
+
+```
+./scripts/bootstrap-rootfs-overlay-demo.sh --tenant-token <paste token from Hosted Mender>
```

Comment:
Mainly for cognitive reasons, I'm wondering if an `-o` output directory argument should be mandatory for this script. The reason is that it took me some time to understand the relationship between this script and the next command. With a mandatory output directory it would have been immediately obvious.

Reply:
Thanks, updated based on your suggestion